### PR TITLE
docs: remove warnings, part 1

### PR DESCRIPTION
**Pull request by jchavarri**
Part of #693.

Still a long way to go, there are lots of other warnings in compiler libs, e.g.

```
File "vendor/melange-compiler-libs/lib/parse.mli", line 87, characters 29-43:
Warning: Failed to resolve reference unresolvedroot(Longident).t Couldn't find "Longident"
```

And also in melange itself, due to the way `Js_internal` is included I believe:

```
Warning, resolved hidden path: Js__.Js_internal.Fn.arity2
```

There are also other warnings that I have no clue how to solve about alerts, e.g.:

```
File "jscomp/stdlib/moreLabels.mli", line 63, character 2 to line 65, character 3:
Warning: Alert unsynchronized_access not expected here.
```

But this PR already includes some fixes, maybe can be considered to merge to keep doing progressive improvements.

### CODE REVIEW

File: jscomp/ext/ext_array.mli [hunk 1]
```diff
@@ -23,7 +23,7 @@
  * Foundation, Inc., 59 Temple Place - Suite 330, Boston, MA 02111-1307, USA. *)
 
 val reverse_range : 'a array -> int -> int -> unit
-(** Some utilities for {!Array} operations *)
```
Comment (anmonteiro) [inline jscomp/ext/ext_array.mli:26]: what's the issue here? Does it not link to `Stdlib.Array`?

File: jscomp/others/belt_Id.mli [hunk 2]
```diff
@@ -66,7 +66,7 @@ type ('key, 'id) comparable =
     Unlike normal functions, when created, it comes with a unique identity (guaranteed
     by the type system).
 
-    It can be created using function {!comparableU} or{!comparable}.
+    It can be created using function {!comparableU} or{!val-comparable}.
```
Comment (anmonteiro) [inline jscomp/others/belt_Id.mli:69]: what does `!{val-FOO}` do?

Comment (anmonteiro) [inline jscomp/others/belt_Id.mli:69]:
```suggestion
    It can be created using function {!comparableU} or {!val-comparable}.
```

**Comment by anmonteiro** [comment]
- re: `Js_internal` stuff, I think we can probably add some `@canonical` annotations, e.g. 
https://github.com/ocaml-multicore/eio/blob/19c42eb7a4b763c6307ccf2400624eb25f9b732a/lib_eio/eio_mutex.mli#L13

- I think we can remove some of the `unsynchronized_access`alerts from our stdlib

**Comment by jchavarri** [comment]
thank you sir 🎩

**Pull request merged**